spec/domain/export/ics/events_spec.rb

```diff
@@ -32,6 +29,60 @@
     end
   end
 
+  describe '#generate_ical_from_event_date' do
+    subject(:ical_event) { export.generate_ical_from_event_date(event_date, event) }
+
+    context 'with only a start date' do
+      let(:event_date) do
+        Event::Date.new(event: event, start_at: Time.zone.local(2018, 5, 19), location: 'testlocation')
+      end
+
+      it do
+        expect(ical_event.dtstart).to be_a(ical_date_klass)
+        expect(ical_event.dtstart.value_ical).to eq(event_date.start_at.strftime(ical_date_klass::FORMAT))
+        expect(ical_event.dtend).to be nil
+        expect(ical_event.summary.to_s).to eq("#{event.name}: #{event_date.label}")
+        expect(ical_event.location.to_s).to eq(event_date.location)
+      end
+    end
+
+    context 'with a start datetime and an end datetime' do
+      let(:event_date) do
+        Event::Date.new(
+          event: event,
+          start_at: Time.zone.local(2018, 5, 19, 12, 0),
+          finish_at: Time.zone.local(2018, 5, 21, 16, 0)
+        )
+      end
+
+      it do
+        expect(ical_event.dtstart).to be_a(ical_datetime_klass)
+        expect(ical_event.dtstart.value_ical).to eq(event_date.start_at.strftime(ical_datetime_klass::FORMAT))
+        expect(ical_event.dtend).to be_a(ical_datetime_klass)
+        expect(ical_event.dtend).to eq(event_date.finish_at.strftime(ical_datetime_klass::FORMAT))
+      end
+    end
+  end
+
+  describe '#datetime_to_ical' do
+    subject { export.datetime_to_ical(datetime) }
+
+    context 'with fullday event' do
+      let(:datetime) { Time.zone.local(2018, 5, 19) }
+      it { is_expected.to be_a(Icalendar::Values::Date) }
```

Comment:
I would use the `let` from above to have the class itself mentioned in fewer places.

Reply:
👍